File: fixtures/typescript/object_method_duplicate/right.ts
export const invoiceTools = {
  buildLines(invoices: Invoice[]): Line[] {
    const shown = invoices.filter((invoice) => invoice.shown);
    const lines = shown.map((invoice) => ({ key: invoice.key, amount: invoice.amount }));
    const active = lines.filter((line) => line.amount > 0);
    return active;
  },
};

File: fixtures/mixed/rust_and_typescript/right.rs
pub fn build_invoice_rows(invoices: &[Invoice]) -> Vec<Line> {
    let shown = invoices.iter().filter(|invoice| invoice.shown).collect::<Vec<_>>();
    let lines = shown.iter().map(|invoice| invoice.to_line()).collect::<Vec<_>>();
    let active = lines.into_iter().filter(|line| line.amount > 0).collect::<Vec<_>>();
    active
}
